huggingface_pipelines/dataset.py

```diff
@@ -0,0 +1,137 @@
+        output_dir (str): The directory to store the output datasets.
+        streaming (bool): Whether to stream the dataset or load it entirely into memory.
+        config (str): The specific configuration of the dataset to load, if applicable.
+        world_size (int): The total number of shards to split the dataset into.
+        rank (int): The index of the shard to retrieve (0-based).
+        trust_remote_code (bool): Whether to trust remote code when loading the dataset.
+
+    Note:
+        The `world_size` and `rank` attributes are particularly useful for distributed data processing,
+        allowing the dataset to be split across multiple processes or machines.
+    """
+
+    dataset_name: str
+    dataset_split: str
+    output_dir: str
+    streaming: bool = False
+    config: str = None
+    world_size: int = 1
+    rank: int = 0
+    trust_remote_code: bool = False
+
+    def load_dataset(self):
+        """
+        Loads and optionally shards the dataset based on the configuration settings.
+
+        This method uses the Hugging Face datasets library to load the dataset.
+        If `world_size` is greater than 1, it also shards the dataset.
+
+        Returns:
+            datasets.Dataset: The loaded and potentially sharded dataset.
+
+        Raises:
+            ValueError: If the dataset cannot be loaded with the given configuration.
+            ImportError: If the 'datasets' library is not installed.
+
+        Note:
+            Ensure that the 'datasets' library is installed before calling this method.
+        """
+        from datasets import load_dataset
+
+        dataset_kwargs = self.get_dataset_kwargs()
+        dataset = load_dataset(**dataset_kwargs)
+
+        self.validate_world_size_and_rank()
+
+        if not self.streaming and self.world_size > 1:
+            dataset = dataset.shard(
+                num_shards=self.world_size, index=self.rank)
+
+        return dataset
+
+    def get_dataset_kwargs(self) -> dict[str, Any]:
+        """
+        Returns the kwargs for load_dataset function.
+
+        This method prepares the keyword arguments used in the load_dataset function.
+
+        Returns:
+            dict[str, Any]: A dictionary of keyword arguments for load_dataset.
+        """
+        dataset_kwargs = {
+            "path": self.dataset_name,
+            "name": self.config,
+            "split": self.dataset_split,
+            "trust_remote_code": self.trust_remote_code,
+            "streaming": self.streaming
+        }
+        return {k: v for k, v in dataset_kwargs.items() if v is not None}
+
+    def validate_world_size_and_rank(self):
+        """
+        Validates world_size and rank.
+
+        Raises:
+            AssertionError: If world_size or rank are invalid.
+        """
+        assert self.world_size >= 1, f"Invalid world_size: {self.world_size}. It should be >= 1."
```

Comment:
nit comment, but this might help you in the future:

in standard SWE, `assert` is normally to assert low-level, internal variables that should act in certain way in normal cases, to define the points of fail fast in the program.

When it comes to validating user inputs, it is normally better to use "if" condition and raise ValueError